src/gui/CMakeLists.txt

```diff
@@ -54,6 +54,17 @@ if (Qt5_FOUND AND BUILD_GUI)
       Boost::boost
   )
 
+  # add custom call to swig to generate required header
+  add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/tclSwig.h
+    COMMAND ${SWIG_EXECUTABLE} -tcl -external-runtime ${CMAKE_CURRENT_BINARY_DIR}/tclSwig.h
+  )
+  set_property(
+    SOURCE src/tclCmdInputWidget.cpp
+    APPEND PROPERTY INCLUDE_DIRECTORIES ${CMAKE_CURRENT_BINARY_DIR}
+    APPEND PROPERTY OBJECT_DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/tclSwig.h
+  )
+
```

Comment:
Can this be folded into src/cmake/swig_lib.cmake as an option?

Reply:
I don't see why not. Would you want to pass in the source files that need it and move this whole set into swig_lib? 
like:
```
 swig_lib(NAME      gui
           NAMESPACE gui
           I_FILE    src/gui.i
           RUNTIME_HEADER tclSwig.h
           RUNTIME_DEPENDS src/tclCmd...
  )
```
or just the generation of the runtime header?
```
swig_lib(NAME      gui
           NAMESPACE gui
           I_FILE    src/gui.i
           RUNTIME_HEADER tclSwig.h
  )
```
and keep the set_property in the gui?

Reply:
I think just RUNTIME_HEADER but I would use this instead of the property (inside swib_lib):

    target_include_directories(${ARG_NAME}
      PRIVATE
        ${CMAKE_CURRENT_BINARY_DIR}
    )

The dependency should be handled by the normal include scanning.